src/main/java/jackyy/dimensionaledibles/block/BlockCakeBase.java

```diff
@@ -112,7 +112,7 @@ public boolean onBlockActivated(World worldIn,
 
         ItemStack stack = playerIn.getHeldItem(hand);
         if (!stack.isEmpty() &&
-            stack.getItem() == Item.REGISTRY.getObject(new ResourceLocation(config().fuel())) &&
+            ItemStack.areItemsEqual(stack, getFuelItemStack()) &&
```

Comment:
This method does not compare itemstack NBT. Would this be something that we would want to support for fuel item?

Reply:
It would be simple enough to check for it, though I'll leave that decision up to someone else

Reply:
Supporting NBT would be a new feature entirely and I think it would be outside the scope of this code cleanup.